scripts/environment_json.mjs

```diff
@@ -27,15 +27,23 @@ import minimist from "minimist";
 */
 export async function environmentJson(platform, buildMode) {
   if (!platform) {
-    throw new Error("environmentJson requires a platform argument");
+    throw new TypeError("environmentJson requires a platform argument");
   }
 
   if (!(buildMode === "debug" || buildMode === "release")) {
-    throw new Error("environmentJson requires a buildMode argument of either 'debug' or 'release'");
+    throw new TypeError("environmentJson requires a buildMode argument of either 'debug' or 'release'");
   }
 
-  if (buildMode === "release" && !process.env.SENTRY_DSN) {
-    throw new Error("Release builds require SENTRY_DSN, but it is not defined.");
+  if (buildMode === "release") {
+    if (!process.env.SENTRY_DSN) {
+      throw new TypeError("Release builds require SENTRY_DSN, but it is not defined.");
+    }
+    
+    try {
+      new URL(process.env.SENTRY_DSN);
```

Comment:
this will throw if the string is bad:

```
~/outline/outline-client % node                                                           
Welcome to Node.js v16.10.0.
Type ".help" for more information.
> new URL("test")
Uncaught TypeError [ERR_INVALID_URL]: Invalid URL
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at new NodeError (node:internal/errors:371:5)
    at onParseError (node:internal/url:552:9)
    at new URL (node:internal/url:628:5) {
  input: 'test',
  code: 'ERR_INVALID_URL'
}
```

Reply:
You might want to link to the format documentation: https://docs.sentry.io/product/sentry-basics/dsn-explainer/#the-parts-of-the-dsn

Reply:
Sure, done